src/lib.rs

```diff
@@ -390,7 +388,13 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>> HotShot<TYPES::ConsensusType
         async_spawn_local(async move {
             if inner
                 .networking
-                .broadcast_message(Message { sender: pk, kind })
+                .broadcast_message_cc(
+                    Message { sender: pk, kind },
+                    // TODO this is morally wrong
```

Comment:
Why? Because of the `clone()`? 😄

Reply:
I think now this is assuming that Election is threadsafe but not putting it behind a rwlock or mutex (maybe I should do that?).

Reply:
Would it be simpler to make `braodcast_message` take the committee set (i.e., the `BTreeSet` retrieved from `get_committee(election, view_number)`) rather than the whole election as input to avoid the thread safety problem? That's the only info needed for message broadcasting.

Reply:
This is a good point! I think this is simpler.

Long term, my understanding is we'll need more from election than just the keyset. Notably, the DA committee won't be fixed, so we'll probably want to query a read-only version of the Election every time we need a few more keys. Maybe this could be a stream? Or some sort iterator of keys? @nyospe are there other things that broadcast might need from election? If not, I'll switch this back to a keyset, and add Election as maybe an associated type? We still would need to use the Election generic somewhere.

Reply:
Yeah sounds good to me. If we are going to query more items from `Election`, it makes sense to have a read-only version.

Reply:
Yeah, this makes sense!